### ISSUE TITLE
na_ontap_user_role not working correctly in REST mode

### ISSUE THREAD

**Issue by mamoep** [ansible-collections/netapp.ontap#111]
### Summary

I am using na_ontap_user_role to create/modify roles. My playbook fails with collection version >= 21.23.0

### Component Name

na_ontap_user_role

### Ansible Version

```console
$ ansible --version
ansible 2.9.27
  config file = /home/user/.ansible.cfg
  configured module search path = [u'/home/user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, May 27 2022, 11:27:32) [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
```

### ONTAP Collection Version

```console
$ ansible-galaxy collection list
collection version tested: 
- 21.23.0
- 21.24.1
```

### ONTAP Version

```console
NetApp Release 9.7P19: Mon May 23 20:55:08 UTC 2022
NetApp Release 9.9.1P9: Wed Apr 20 00:42:19 UTC 2022
```

### Playbook

```console
- hosts: localhost
  become: false
  gather_facts: false
  connection: local
  collections:
    - netapp.ontap
  name: ONTAP Debug
  vars:
    login: &login
      hostname: "clustername"
      username: "{{ username }}"
      password: "{{ password }}"
      https: true
      validate_certs: true
  tasks:
  - name: test user role
    na_ontap_user_role:
      state: present
      vserver: "clustername"
      name: operator
      command_directory_name: "network interface migrate"
      access_level: all
      <<: *login
```

### Steps to Reproduce

```yaml (paste below)

```
Execute example playbook. Setting "use_rest: never" works as a workaround.

### Expected Results

Successful run of the playbook

### Actual Results

```console
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: None
  localhost failed | msg: Error creating role operator: calling: security/roles: got {u'message': u'Invalid JSON input. Expecting "privileges" to be an array.', u'code': u'262254', u'target': u'privileges.access'}.
```

**Comment by lonico** [comment]
I think you need to set privileges.   Can you try that as a work-around?

On our side we need to validate privileges and its contents, or build a default.

**Comment by mamoep** [comment]
Sorry I don‘t get. My playbook worked fine until REST support was added to the module. I don‘t understand what privileges is for and it is not very well documented.

**Comment by carchi8py** [comment]
@mamoep add this to your playbook

use_rest: Never

Should switch you back to Zapi and avoid the issue while we fix the bug

**Comment by lonico** [comment]
DEVOPS-5626

**Comment by lonico** [comment]
fixed in 22.0.0
